test/unit/gateways/first_pay_json_test.rb

```diff
@@ -0,0 +1,454 @@
+require 'test_helper'
+
+class FirstPayJsonTest < Test::Unit::TestCase
+  include CommStub
+
+  def setup
+    @gateway = FirstPayJsonGateway.new(
+      processor_id: 1234,
+      merchant_key: 'a91c38c3-7d7f-4d29-acc7-927b4dca0dbe'
+    )
+
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      order_id: SecureRandom.hex(24),
+      billing_address: address
+    }
+  end
+
+  def test_successful_purchase
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(/transactionAmount/, data)
+      assert_match(/cardNumber/, data)
+      assert_match(/cardExpMonth/, data)
+      assert_match(/cardExpYear/, data)
+      assert_match(/cvv/, data)
+      assert_match(/ownerName/, data)
+      assert_match(/ownerStreet/, data)
+      assert_match(/ownerCity/, data)
+      assert_match(/ownerState/, data)
+      assert_match(/ownerZip/, data)
+      assert_match(/ownerCountry/, data)
+      assert_match(/processorId/, data)
+      assert_match(/merchantKey/, data)
+    end.respond_with(successful_purchase_response)
+
+    assert response
+    assert_instance_of Response, response
+    assert_success response
+    assert_equal '31076534', response.authorization
+    assert_equal 'Approved 735498', response.message
+  end
+
+  def test_failed_purchase
```

Comment:
💬 Comment / Style:

Is going to give the same result but just to follow the patterns, `sub_comms` methods default behavior is to stub the :ssl_post, so this can also be written like:

```ruby
  def test_failed_purchase
    response = stub_comms do
      @gateway.purchase(200, @credit_card, @options)
    end.respond_with(failed_purchase_response)

    assert response
    assert_instance_of Response, response
    assert_failure response
    assert_equal '31076656', response.authorization
    assert_equal 'Auth Declined', response.message
  end

```